include/deal.II/arborx/bvh.h

```diff
@@ -0,0 +1,134 @@
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_arborx_bvh_h
+#define dealii_arborx_bvh_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_ARBORX
+#  include <deal.II/arborx/access_traits.h>
+
+#  include <ArborX_LinearBVH.hpp>
+#  include <Kokkos_Core.hpp>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This namespace contains wrappers for the ArborX library.
+ */
+namespace ArborXWrappers
+{
+  /**
+   * This class implements a wrapper around ArborX::BVH. BVH stands for Bounding
+   * Volume Hierarchy.
+   *
+   * From [Wikipedia](https://en.wikipedia.org/wiki/Bounding_Volume_Hierarchy):
+   * <blockquote>
+   * A bounding volume hierarchy (BVH) is a tree structure on a set of geometric
+   * objects. All geometric objects are wrapped in bounding volumes that form
+   * the leaf nodes of the tree. These nodes are then grouped as small sets and
+   * enclosed within larger bounding volumes. These, in turn, are also grouped
+   * and enclosed within other larger bounding volumes in a recursive fashion,
+   * eventually resulting in a tree structure with a single bounding volume at
+   * the top of the tree. Bounding volume hierarchies are used to support
+   * several operations on sets of geometric objects efficiently, such as in
+   * collision detection and ray tracing.
+   * </blockquote>
+   *
+   * Because ArborX uses Kokkos, Kokkos needs to be initialized and finalized
+   * before using this class.
+   */
+  class BVH
+  {
+  public:
+    /**
+     * Constructor. Use a vector of BoundingBox @p bounding_boxes as primitives.
+     */
+    template <int dim, typename Number>
+    BVH(const std::vector<BoundingBox<dim, Number>> &bounding_boxes);
+
+    /**
+     * Return the indices of the BoundingBox that satisfies the @p queries.
+     * Because @p queries can contain multiple queries, the function returns a pair
+     * of indices and offsets. The code to get the intersections of bounding
+     * boxes with the bounding boxes from the BVH looks like this:
+     *
+     * @code
+     * std::vector<BoundingBox<dim>> query_bounding_boxes =
+     * get_query_bounding_boxes();
+     * ArborXWrappers::BoundingBoxIntersect bb_intersect(query_bounding_boxes);
+     *
+     * std::vector<BoundingBox<dim>> bvh_bounding_boxes =
+     * get_bvh_bounding_boxes();
+     * ArborxWrappers::BVH bvh(bvh_bounding_boxes);
+     *
+     * auto [indices, offset] = bvh.query(bb_intersect);
+     * @endcode
+     *
+     * The indices that satisfy the first query are given by:
+     *
+     * @code
+     * std::vector<int> first_query_indices;
+     * for (int i = offset[0]; i < offset[1]; ++i)
+     *   first_query_indices.push_back(indices[i]);
+     * @endcode
+     */
```

Comment:
The example above does a query for bounding boxes. Could you add a short similar example also for the point query? Maybe with a text such as this:

In many other applications, we are interested not only in finding which bounding boxes another bounding box lies in, but in fact which bounding boxes individual points lie in -- say, if instead of objects we have point-like particles moving around. In that case, we would need to answer a query for points, and this can be done as follows: